src/main/java/org/jabref/gui/maintable/NewLibraryFromPdfAction.java

```diff
@@ -0,0 +1,99 @@
+ *
+ * <ul>
+ *   <li>Mode choice A: online or offline</li>
+ *   <li>Mode choice B: complete entry or single file (the latter is not implemented)</li>
+ * </ul>
+ * <p>
+ * The mode is selected by the preferences whether to use Grobid or not.
+ * <p>
+ * The different modes should be implemented as sub classes. Moreover, there are synergies with {@link ExtractReferencesAction}. However, this was too complicated.
+ */
+public class NewLibraryFromPdfAction extends SimpleCommand {
+    private static final Logger LOGGER = LoggerFactory.getLogger(NewLibraryFromPdfAction .class);
+
+    private final LibraryTabContainer libraryTabContainer;
+    private final StateManager stateManager;
+    private final DialogService dialogService;
+    private final PreferencesService preferencesService;
+    private final BibliographyFromPdfImporter bibliographyFromPdfImporter;
+    private final TaskExecutor taskExecutor;
+
+    public NewLibraryFromPdfAction(
+            LibraryTabContainer libraryTabContainer,
+            StateManager stateManager,
+            DialogService dialogService,
+            PreferencesService preferencesService,
+            TaskExecutor taskExecutor) {
+        this.libraryTabContainer = libraryTabContainer;
+        this.stateManager = stateManager;
+        this.dialogService = dialogService;
+        this.preferencesService = preferencesService;
+        // Instruct the importer to keep the numbers (instead of generating keys)
+        this.bibliographyFromPdfImporter = new BibliographyFromPdfImporter();
+        this.taskExecutor = taskExecutor;
+    }
+
+    @Override
+    public void execute() {
+        final FileDialogConfiguration.Builder builder = new FileDialogConfiguration.Builder();
+        builder.withDefaultExtension(StandardFileType.PDF);
+        // Sensible default for the directory to start browsing is the directory of the currently opened library. The pdf storage dir seems not to be feasible, because extracting references from a PDF itself can be done by the context menu of the respective entry.
+        stateManager.getActiveDatabase()
+                    .flatMap(db -> db.getDatabasePath())
+                    .ifPresent(path -> builder.withInitialDirectory(path.getParent()));
+        FileDialogConfiguration fileDialogConfiguration = builder.build();
+
+        LOGGER.trace("Opening file dialog with configuration: {}", fileDialogConfiguration);
+
+        dialogService.showFileOpenDialog(fileDialogConfiguration).ifPresent(path -> {
+            LOGGER.trace("Selected file: {}", path);
+            Callable<ParserResult> parserResultCallable = getParserResultCallable(path);
+            BackgroundTask.wrap(parserResultCallable)
+                          .withInitialMessage(Localization.lang("Processing PDF(s)"))
+                          .onFailure(failure -> Platform.runLater(() -> dialogService.showErrorDialogAndWait(failure)))
```

Comment:
onFailure  and onSucccces are alwas execute on the fx thraed, no wrapping necessary